src/server/game/Spells/SpellEffects.cpp

```diff
@@ -4708,9 +4708,10 @@ void Spell::EffectKnockBack(SpellEffIndex effIndex)
     if (!unitTarget)
         return;
 
-    if (Creature* creatureTarget = unitTarget->ToCreature())
-        if (creatureTarget->isWorldBoss() || creatureTarget->IsDungeonBoss())
-            return;
+    if (m_caster->GetTypeId() == TYPEID_PLAYER || m_caster->GetOwnerGUID().IsPlayer() || m_caster->IsHunterPet())
```

Comment:
@boxa MECHANIC_KNOCKOUT is not KNOCKBACK - it is rogue gouge and engineering bombs that break on damage taken (so your code is actually wrong)